File: surf-chat/src/main/java/dev/slne/surf/social/chat/command/channel/ChannelStateToggleCommand.java
package dev.slne.surf.social.chat.command.channel;

import dev.jorel.commandapi.CommandAPICommand;
import dev.slne.surf.social.chat.SurfChat;
import dev.slne.surf.social.chat.object.Channel;
import dev.slne.surf.social.chat.util.MessageBuilder;

public class ChannelStateToggleCommand extends CommandAPICommand {

  public ChannelStateToggleCommand(String commandName) {
    super(commandName);

    executesPlayer((player, args) -> {
      Channel channel = Channel.getChannel(player);

      if(channel == null) {
        return;
      }

      if(!channel.isOwner(player)) {
        return;
      }

     if(channel.isClosed()) {
       channel.setClosed(false);
       SurfChat.message(player, new MessageBuilder().primary("Der Nachrichtenkanal ").info(channel.getName()).primary(" ist nun ").success("öffentlich."));
     } else {
       channel.setClosed(true);
       SurfChat.message(player, new MessageBuilder().primary("Der Nachrichtenkanal ").info(channel.getName()).primary(" ist nun ").error("privat."));
     }
    });
  }
}

File: surf-chat/src/main/java/dev/slne/surf/social/chat/command/channel/ChannelStateToggleCommand.kt
package dev.slne.surf.social.chat.command.channel

import dev.jorel.commandapi.CommandAPICommand
import dev.jorel.commandapi.executors.CommandArguments
import dev.jorel.commandapi.executors.PlayerCommandExecutor
import dev.jorel.commandapi.kotlindsl.playerExecutor
import dev.slne.surf.social.chat.SurfChat
import dev.slne.surf.social.chat.`object`.Channel
import dev.slne.surf.social.chat.send
import dev.slne.surf.social.chat.util.MessageBuilder
import org.bukkit.entity.Player

class ChannelStateToggleCommand(commandName: String) : CommandAPICommand(commandName) {
    init {
        playerExecutor { player, _ ->
            val channel: Channel? = Channel.getChannel(player)

            if (channel == null) {
                player.send {
                    appendPrefix()
                    error("Du bist in keinem Nachrichtenkanal.")
                }
                return@playerExecutor
            }

            if (!channel.isOwner(player)) {
                SurfChat.send(player, MessageBuilder().error("Du bist nicht der Besitzer des Nachrichtenkanals."))
                player.send {
                    appendPrefix()
                    error("Du bist nicht der Besitzer des Nachrichtenkanals.")
                }
                return@playerExecutor
            }

            if (channel.closed) {
                channel.closed = false
                player.send {
                    appendPrefix()
                    primary("Der Nachrichtenkanal ")
                    info(channel.name)
                    primary(" ist nun ")
                    success("öffentlich.")
                }
            } else {
                channel.closed = true
                player.send {
                    appendPrefix()
                    primary("Der Nachrichtenkanal ")
                    info(channel.name)
                    primary(" ist nun ")
                    error("privat.")
                }
            }
        }
    }
}
